azurelinuxagent/common/protocol/wire.py

```diff
@@ -671,6 +677,17 @@ def update_goal_state(self, forced=False, max_retry=3):
                 self.update_shared_conf(goal_state)
                 self.update_certs(goal_state)
                 self.update_ext_conf(goal_state)
+
+                if conf.enable_rdma():
+                    if rdma_device_configured:
+                        logger.info("RDMA device already configured, skipping")
+                    else:
+                        logger.info("RDMA capabilities are enabled in configuration")
+                        rdma_device_configured = True # do not run again
+                        self.setup_rdma_dev()
```

Comment:
@hglkrijger if we get another call to Goal State code path, then we may run into race condition where we have two threads writing to the /dev/hvnd_rdma device.

RDMA flow is unfortunately just fragile and if something fails here it is unlikely that it will pass second time. I'm not sure flipping this will do any good to us.